mysite/chat/serializers.py

```diff
@@ -0,0 +1,298 @@
+import injections
+from rest_framework import serializers
+from django.core.urlresolvers import reverse
+
+from .models import Message, Chat, ChatDivider
+from .services import ProgressHandler
+from ct.models import UnitLesson, Response
+from accounts.models import Instructor
+
+
+class InternalMessageSerializer(serializers.ModelSerializer):
+    """
+    Serializer for addMessage list representation.
+    """
+    html = serializers.CharField(source='get_html', read_only=True)
+    name = serializers.CharField(source='get_name', read_only=True)
+    avatar = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Message
+        fields = (
+            'id',
+            'type',
+            'name',
+            'userMessage',
+            'avatar',
+            'html'
+        )
+
+    def get_avatar(self, obj):
+        if not obj.userMessage:
+            try:
+                return obj.chat.instructor.instructor.icon_url
+            except Instructor.DoesNotExist:
+                pass
+
+
+class InputSerializer(serializers.Serializer):
+    """
+    Serializer for input description for next message.
+    """
+    type = serializers.CharField(max_length=16, read_only=True)
+    url = serializers.CharField(max_length=64, read_only=True)
+    options = serializers.ListField()
+    includeSelectedValuesFromMessages = serializers.ListField(
+        child=serializers.IntegerField(min_value=0)
+    )
+    html = serializers.CharField(max_length=300, read_only=True)
+
+
+@injections.has
+class MessageSerializer(serializers.ModelSerializer):
+    """
+    Message serializer.
+    """
+    next_handler = injections.depends(ProgressHandler)
+
+    input = serializers.SerializerMethodField()
+    # errors = serializers.CharField(source='get_errors', read_only=True)
+    addMessages = serializers.SerializerMethodField()
```

Comment:
I don't see the necessity of declaring `input` and `addMessages` field's serialization. As this is a model field, specifying fields in Meta class is enough, I believe.